### PR TITLE
Updated dev container CI

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
1. Setting tag to a non-versioned "latest" given that the container is always built manually
  - Dev Containers from previous versions of Dapr are [failing to build](https://github.com/dapr/dapr/actions/workflows/dapr-dev-container.yml) right now (until 1.8, the Action had an automatic trigger and we had to turn that off as it was causing issues like overriding the dev container on every hotfix)
  - We often make changes in the "master" branch such as upgrading the version of Go or the linter being used, so we should be able to re-publish dev containers whenever we feel like we should
2. Updated the Action to upgrade dependencies that will be deprecated soon ([see warnings here](https://github.com/dapr/dapr/actions/runs/3606497150))

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
 	"name": "Dapr Dev Environment",
 	// Update the container version when you publish dev-container
-	"image": "ghcr.io/dapr/dapr-dev:0.1.9",
+	"image": "ghcr.io/dapr/dapr-dev:latest",
```
Comment (daixiang0) [inline .devcontainer/devcontainer.json:4]: Now does it auto-build and push?

**Review by shubham1172 — DISMISSED** [review]

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (ItalyPaleAle) [reply .devcontainer/devcontainer.json:4]: https://github.com/dapr/dapr/actions/runs/3606497150/workflow#L16

Comment (daixiang0) [reply .devcontainer/devcontainer.json:4]: Although we have updated “build dev container” workflow, but for developers, they can not get those changes, right?

Could we re-enable this job?

Comment (ItalyPaleAle) [reply .devcontainer/devcontainer.json:4]: What do you mean? The job is enabled, just needs to be triggered manually

Comment (ItalyPaleAle) [reply .devcontainer/devcontainer.json:4]: Done. It now runs weekly too

**Review by shubham1172 — DISMISSED** [review]

File: .github/workflows/dapr-dev-container.yml [hunk 2]
```diff
@@ -15,11 +15,14 @@ name: dapr-dev-container
 
 on:
   workflow_dispatch:
+  schedule:
+    # Run weekly on Tuesdays
+    - cron: "22 03 * * 2"
```
Comment (shubham1172) [inline .github/workflows/dapr-dev-container.yml:20]: What's with 3:22 though? 😄

Comment (ItalyPaleAle) [reply .github/workflows/dapr-dev-container.yml:20]: A random time that is not on top of the hour (when a lot of concurrent tasks are happening), and that’s in the middle of the night PST, when we see less activity on the repo.
I actually put a bit of thought on this :)

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (mukundansundar) [reply .devcontainer/devcontainer.json:4]: If its being triggered automatically, can this create an issue if we are always going to use the latest dev container?

Comment (DeepanshuA) [reply .devcontainer/devcontainer.json:4]: I think mostly developers would be using it, so may be that is better if we get latest.

Comment (ItalyPaleAle) [reply .devcontainer/devcontainer.json:4]: What kind of issues are you concerned with?

**Review by DeepanshuA — APPROVED** [review]

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (artursouza) [reply .devcontainer/devcontainer.json:4]: Can we avoid using latest?

File: .github/workflows/dapr-dev-container.yml (hunk 2 above)
Comment (artursouza) [reply .github/workflows/dapr-dev-container.yml:20]: Genesis 3:22

And the Lord God said, “The man has now become like one of us, knowing good and evil. He must not be allowed to reach out his hand and take also from the tree of life and eat, and live forever.”

File: docker/docker.mk [hunk 3]
```diff
@@ -242,7 +242,7 @@ docker-windows-base-push: check-windows-version
 ################################################################################
 
 # Update whenever you upgrade dev container image
-DEV_CONTAINER_VERSION_TAG?=0.1.9
```
Comment (artursouza) [inline docker/docker.mk:245]: Please, avoid using `latest` - we got hit by this in multiple places before.

**Review by artursouza — CHANGES_REQUESTED** [review]
Avoid `latest` unless it is needed for sure.

**Comment by ItalyPaleAle** [comment]
> Please, avoid using latest - we got hit by this in multiple places before.

What were the issues before?

The reason for `latest` is actually because "master" (which is where we should assume most contributors are working in) moves fast and we should push images quickly.

Note that for the last few months, dev containers have been built manually, so hotfixes wouldn't override older containers. In the past, they were built on every tag. This meant that when 1.8.0 was released, dev container 1.8 was built. However, "master" branch evolved, and we (for example) upgraded Go. We re-triggered the dev container so 1.8 was re-built. Everything was fine until 1.8.1 came out, and then the dev container was rebuilt with the old version of Go. This is why we've moved months ago to only triggering the dev container manually.

**Comment by berndverst** [comment]
While you are touching things related to the dev container, can you please enable the SSH feature in the dev container json?

```
"features": {
    "sshd": "latest"
}
```

**Comment by ItalyPaleAle** [comment]
@berndverst done

**Review by artursouza — APPROVED** [review]

**Pull request merged**